src/terminal/adapter/DispatchTypes.hpp

```diff
@@ -109,31 +111,33 @@ namespace Microsoft::Console::VirtualTerminal
             return _value >= 0;
         }
 
-        constexpr size_t value() const noexcept
+        constexpr VTInt value() const noexcept
         {
             return _value;
         }
 
-        constexpr size_t value_or(size_t defaultValue) const noexcept
+        constexpr VTInt value_or(VTInt defaultValue) const noexcept
         {
-            return has_value() ? _value : defaultValue;
+            // A negative value indicates that the parameter was omitted.
+            return _value < 0 ? defaultValue : _value;
         }
 
-        template<typename T, std::enable_if_t<sizeof(T) == sizeof(size_t), int> = 0>
+        template<typename T, typename = std::enable_if_t<std::is_enum_v<T>>>
         constexpr operator T() const noexcept
         {
             // For most selective parameters, omitted values will default to 0.
             return static_cast<T>(value_or(0));
         }
 
-        constexpr operator size_t() const noexcept
+        constexpr operator VTInt() const noexcept
         {
             // For numeric parameters, both 0 and omitted values will default to 1.
-            return has_value() && _value != 0 ? _value : 1;
+            // The parameter is omitted if _value is less than 0.
+            return _value <= 0 ? 1 : _value;
         }
 
     private:
-        std::make_signed<size_t>::type _value;
+        VTInt _value;

```

Comment:
This is the part that scares me -- I suspect, however, that @j4james authored tests making sure that conversions to enum vs integer work properly :)

Reply:
I'd agree with reverting these changes.
But generally, it used to convert `intptr_t` to enums whose underlying type is `size_t`, and the new code converts `int` to `int`, so we should be fine, if not "finer" than before.